hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieClient.java

```diff
@@ -284,4 +288,33 @@ protected void writeTableMetadata(HoodieTable table, String instantTime, HoodieC
       }
     }
   }
+
+  public static void updateColsToIndex(HoodieTable dataTable, HoodieWriteConfig config, HoodieCommitMetadata commitMetadata) {
```

Comment:
Though it is safe to update in multi-writer scenario as this is still called within a transaction, I am thinking should we do all table config updates in one shot instead opening and closing the file multiple times? Let's say if this was the first commit building the colstats, then table config would be update multiple times, isn't it?